packages/sanity/src/core/store/_legacy/history/history/Timeline.ts

```diff
@@ -194,7 +194,11 @@ export class Timeline {
     const nextTransactionToChunk = this._chunks.length > 0 ? this._chunks.last.end : firstIdx
     for (let idx = nextTransactionToChunk; idx <= lastIdx; idx++) {
       const transaction = this._transactions.get(idx)
-      this._chunks.mergeAtEnd(chunkFromTransaction(transaction), mergeChunk)
+      const allTransactions = this._transactions.getAllItems
```

Comment:
To generate the event, we need the previous events, because some of them depend on checks like:
- publishExisted?
- draftExists?

All of this will be more easy to read going to the[ `getEventFromTransaction` function](https://github.com/sanity-io/sanity/pull/7569/files#diff-8260f1c6aa91c7581f94b96f7b21931b038475e0d866b6317e1961e26e395f7d)